src/anm/events.js

```diff
@@ -164,28 +164,31 @@ MouseEventsSupport.prototype.dispatch = function(event) {
         return true;
     }
 
-    var dispatchedChild;  // found the matching child inside
-    if (owner.inside(localEvent)) {
-        owner.reverseEach(function(child) {
-            if (child.isActive()) {
-                dispatchedChild = child.dispatchMouseEvent(localEvent) ? child : null;
-                if (dispatchedChild) return false; // stop iteration of reverseEach
-            }
-        });
+    var dispatchedByOwner, // handled event myself
+        dispatchedByChild; // not handled myself, but found the matching child inside
 
-        if (!dispatchedChild) {
-            if (localEvent.type === 'mousemove') {
-                this.processMove(localEvent); // fire mouseenter/mouseexit if required
-            }
-            if ((localEvent.type === 'mouseclick') || (localEvent.type === 'mousedown')) {
-                state.pressedNode = this.owner;
-            }
-            this.owner.fire(localEvent.type, localEvent);
+    // here and below localEvent has properties `.x` and `.y`, so duck typing works
+    dispatchedByOwner = owner.inBounds && owner.inBounds(localEvent) && owner.inside && owner.inside(localEvent);
```

Comment:
usually checks like that are done this way:
i.e. inBounds is internal optimization of inside. for example for image inBounds is not needed at all, it's the same as inside. in fact I suggest to do it first without inBounds at all.

  primitive.inside(p) {
     return inBounds(p) && insidethiselement(p);
  }

  composite.inside(p) {
     if (inBounds(p)) {
        children.each { child => if (child.inside(p)) return true; }
     }
     return false;
  }